util/projectRequest.ts

```diff
@@ -0,0 +1,93 @@
+import { Octokit } from '@octokit/core';
+import { paginateRest } from '@octokit/plugin-paginate-rest';
+import { Project, ACMCommitteeTopics } from './types';
+
+export async function getProjects(): Promise<Project[]> {
+  const PaginatedOctokit = Octokit.plugin(paginateRest);
+  const octokit = new PaginatedOctokit();
+  const projectsResponse = await octokit.paginate('GET /orgs/{org}/repos', {
+    org: 'uclaacm',
+    per_page: 100,
+  });
+  const sortedData = projectsResponse.sort(
+    (a, b) =>
+      new Date(b.updated_at).getTime() - new Date(a.updated_at).getTime(),
+  );
+  return sortedData.map(
+    (repo) =>
+      ({
+        name: repo.name,
+        description: repo.description,
+        link: repo.homepage || null,
+        repo: repo.html_url,
+        lang: repo.language,
+        topics: repo.topics,
+        image: getImageFromTopics(repo.topics).image,
+        alt: getImageFromTopics(repo.topics).alt,
+      } as Project),
+  ) as Project[];
+}
+
+interface ImageInfo {
+  image: string;
+  alt: string;
+}
+
+function topicToImg(topic: string) {
+  switch (topic) {
+    case ACMCommitteeTopics.AI:
+      return {
+        image: '/committee-logos/ai-logo.png',
+        alt: 'ACM AI Logo',
+      } as ImageInfo;
+    case ACMCommitteeTopics.CYBER:
+      return {
+        image: '/committee-logos/cyber-logo.png',
+        alt: 'ACM Cyber Logo',
+      } as ImageInfo;
+    case ACMCommitteeTopics.DESIGN:
+      return {
+        image: '/committee-logos/design-logo.png',
+        alt: 'ACM Design Logo',
+      } as ImageInfo;
+    case ACMCommitteeTopics.HACK:
+      return {
+        image: '/committee-logos/hack-logo.png',
+        alt: 'ACM Hack Logo',
+      } as ImageInfo;
+    case ACMCommitteeTopics.ICPC:
+      return {
+        image: '/committee-logos/icpc-logo.png',
+        alt: 'ACM ICPC Logo',
+      } as ImageInfo;
+    case ACMCommitteeTopics.STUDIO:
+      return {
+        image: '/committee-logos/studio-logo.png',
+        alt: 'ACM Studio Logo',
+      } as ImageInfo;
+    case ACMCommitteeTopics.TEACH_LA:
+      return {
+        image: '/committee-logos/teachla-logo.png',
+        alt: 'ACM Teach LA Logo',
+      } as ImageInfo;
+    case ACMCommitteeTopics.W:
+      return {
+        image: '/committee-logos/w-logo.png',
+        alt: 'ACM W Logo',
+      } as ImageInfo;
+    default:
+      return false;
```

Comment:
Why not have this return the ACM logo, and them simplify the call in `83` to basically just call into this function? Also has the bonus that the function now always returns `ImageInfo`.

Reply:
In `getImageFromTopics`, some topics can be added that aren't an ACM Committee like `learning-labs`, `education`, `react`, etc. I made the default in this case false so that we can iterate through the list of topics until we reach the name of a committee. If we overrode `topicToImg` to return the ACM logo as a default, then we would not properly get the logo of the repository's associated committee if the committee name was not the first topic of the repository.

Reply:
Ah, I see - I think I misunderstood the purpose of this function.

A follow-up question; let's say we add more images in the future that aren't image-based (maybe, for example, we have a special logo for learning labs). You wouldn't be able to add that to this switch-case, right? Since then the order of the topics would matter (and you can't early-exit out of `getImagesFromTopics` once you get the first valid topic). Is this something you think we'll run into?

Reply:
That's a good point! I think that in its current implementation, we can use the `getImagesFromTopics` function as a way to get a default image for each repository, and it would take the order of least precedence from the other ways we can select images/logos for everything. 

I don't think I understand non-image-based logos, since can't all logos be `.png` files as images?

In the future when we implement og:images and social media previews if possible for repos, we can prioritize special images first and use `getImagesFromTopics` as a last-resort if the other ways to get images fail. 

Maybe we should rename it to `getCommitteeImage` instead to better outline its use case?

Reply:
Sorry, image-based is a typo - meant committee-based. Fixed!